operator/validator/event_handler.go

```diff
@@ -123,6 +142,53 @@ func (c *controller) handleValidatorAddedEvent(
 	event abiparser.ValidatorAddedEvent,
 	ongoingSync bool,
 ) ([]zap.Field, error) {
+	_, found, err := c.eventHandler.GetEventData(event.TxHash)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get event data")
+	}
+	if found {
+		// skip
+		return nil, nil
+	}
+
+	// get nonce
+	nonce, err := c.eventHandler.GetNextNonce(event.Owner)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get nonce")
+	}
+
+	err = c.eventHandler.SaveEventData(event.TxHash)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to save event data")
+	}
+
+	err = c.eventHandler.BumpNonce(event.Owner)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to bump the nonce")
+	}
```

Comment:
the saving the txHash and bumping the nonce should happend only after event was handled
By handle I mean:
- 1. the event was handled succuessfully 
- 2. there was malformed event (only!) see https://github.com/bloxapp/ssv/pull/963/files#r1204377224

Reply:
Also the todo comment from last commit was lost
The SaveEventData & BumpNonce shoud be executed as one transaction (atomic).
Not for this PR but lets keep the track of todos